tests/tracer/test_processors.py

```diff
@@ -216,3 +218,79 @@ def test_aggregator_partial_flush_2_spans():
     assert writer.pop() == [child1, child2]
     parent.finish()
     assert writer.pop() == [parent]
+
+
+def test_trace_top_level_span_processor_partial_flushing():
+    """Parent span and child span have the same service name"""
+    tracer = Tracer()
```

Comment:
do we need to use a dummy writer here or something?

Reply:
Yes, right now the tests pass but they log this warning `o handlers could be found for logger ddtrace.internal.writer`

Reply:
probably because it is failing to send the traces and it is trying to log the error.

if we can use a DummyWriter or something to ensure we don't actually try to send these traces it would be best. what if there is an agent running? will the performance/result be different? or we might just send unnecessary test traces to the backend.